### DISCUSSION TITLE
PixiJS on iOS 15 - Broken?

### DISCUSSION THREAD

**Discussion by dglowacki** [pixijs/pixijs#7928] [General]
We're experiencing extreme slowdowns with PixiJS on iOS 15. An order of magnitude slower, and in many cases, it simply doesn't function.

One specific example:
https://pixijs.io/examples/#/masks/graphics.js

On iOS14, this performs well on all devices
On iOS15, this is slow on iPhone 10
On iOS15, this doesn't work on older iPads (2018,2019)

Our games rely on PixiJS, but it appears iOS15 will break our games on downlevel devices entirely.

Anyone else have this experience, and can offer guidance?

**Comment by ivanpopelyshev** [comment]
You can try turn off antialias and use https://github.com/pixijs/graphics-smooth/ instead.

iOS15 is FUBAR

**Comment by photonstorm** [comment]
@dglowacki I would strongly recommend you open an issue on the Webkit bug tracker (to add to the many already there) - here's a similar one from the Construct team: https://bugs.webkit.org/show_bug.cgi?id=230749 - and you can see they've already got an upstream fix in place. The more broken examples they're given (like this) the better for everyone.

**Comment by dglowacki** [comment]
@ivanpopelyshev Thanks for that tip! It saved us from disaster and rebuilding natively.
@photonstorm I'll do that.

**Reply by ivanpopelyshev** [reply]
If you took graphics-smooth , well, that "tip" is actually first ever analytical antialias for production in WebGL. Whether if it works for you, or there's a bug somewhere - please tell us about this experience, its important for later development

**Comment by tywang2006** [comment]
I believe apple is on webkit bug fix now. not sure when the fixed will be released. hopefully it will be soon. this bug affects all web apps  heavily on canvas interactions

**Comment by eXponenta** [comment]
Related #8000
